eduvmstorebackend/eduvmstore/db/models.py

```diff
@@ -1,60 +1,58 @@
-from sqlalchemy import Column, Integer, String, DateTime, Float, Boolean, ForeignKey
-from .session import Base
-from datetime import datetime
+import uuid
 
+from django.db import models
 
-class AppTemplate(Base):
-    __tablename__ = 'app_template'
+from django.utils.timezone import now
 
-    id = Column(String(36), primary_key=True, index=True)
-    image_id = Column(String(36), index=True)
-    name = Column(String(255), nullable=False, unique=True)
-    description = Column(String)
-    short_description = Column(String(255))
-    instantiation_notice = Column(String)
+
+class AppTemplate(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    image_id = models.UUIDField(db_index=True)
+    name = models.CharField(max_length=255, unique=True)
+    description = models.TextField()
+    short_description = models.CharField(max_length=255)
+    instantiation_notice = models.TextField(blank=True, null=True)
 
     # CRUD info
-    creator_id = Column(String(36), ForeignKey('user.id'), index=True)
-    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
-    updated_at = Column(DateTime, default=datetime.utcnow, nullable=True)
-    deleted_at = Column(DateTime, default=datetime.utcnow, nullable=True)
-    deleted = Column(Boolean, default=False)
+    creator_id = models.ForeignKey('User', on_delete=models.DO_NOTHING, db_index=True)
+    created_at = models.DateTimeField(default=now, editable=False)
+    updated_at = models.DateTimeField(default=now)
+    deleted_at = models.DateTimeField()
+    deleted = models.BooleanField(default=False)
 
     # version and visibility
-    version = Column(String, default="1.0")
-    public =Column(Boolean, default=False, nullable=False)
-    approved = Column(Boolean, default=False)
+    version = models.CharField(max_length=50, default="1.0")
+    public = models.BooleanField(default=False)
+    approved = models.BooleanField(default=False)
 
     # resource requirements
-    fixed_ram_gb = Column(Float)
-    fixed_disk_gb = Column(Float)
-    fixed_cores = Column(Integer)
-    per_user_ram_gb = Column(Float)
-    per_user_disk_gb = Column(Float)
-    per_user_cores = Column(Integer)
+    fixed_ram_gb = models.FloatField()
+    fixed_disk_gb = models.FloatField()
+    fixed_cores = models.FloatField()
+    per_user_ram_gb = models.FloatField()
+    per_user_disk_gb = models.FloatField()
+    per_user_cores = models.FloatField()
```

Comment:
gibt es nur FloatField?

Reply:
Float hat wohl Double abgelöst, sonst gibt es halt noch Integer.
Aber Valentin und ich haben diskutiert, dass auch cores Floats sein sollten, um Berechnungen und Angaben in Dezimalform angeben zu können.